src/compiler/checker.ts

```diff
@@ -5124,7 +5153,7 @@ namespace ts {
                         const name = typeParameterToName(newParam, context);
                         const newTypeVariable = factory.createTypeReferenceNode(name);
                         context.approximateLength += 37; // 15 each for two added conditionals, 7 for an added infer type
-                        const newMapper = prependTypeMapping(type.root.checkType, newParam, type.combinedMapper || type.mapper);
+                        const newMapper = prependTypeMapping(type.root.checkType, newParam, type.mapper);

```

Comment:
This line's the only one that was needed to fix the related issue - the rest was just nice refactoring I used to make this change evident while debugging.